accepter.tf

```
@@ -48,46 +48,70 @@ data "aws_vpc" "accepter" {
 }
 
 # Lookup accepter subnets
-data "aws_subnet_ids" "accepter" {
+data "aws_subnets" "accepter" {
   count    = local.accepter_count
   provider = aws.accepter
-  vpc_id   = local.accepter_vpc_id
-  tags     = var.accepter_subnet_tags
+  filter {
+    name   = "vpc-id"
+    values = [local.accepter_vpc_id]
+  }
+  dynamic "filter" {
+    for_each = var.accepter_subnet_tags
+    content {
+      name   = "tag:${filter.key}"
+      values = [filter.value]
+    }
+  }
 }
 
 locals {
-  accepter_subnet_ids       = try(distinct(sort(flatten(data.aws_subnet_ids.accepter.*.ids))), [])
+  accepter_subnet_ids       = try(local.accepter_enabled ? data.aws_subnets.accepter[0].ids : [], [])
```

do we need `try` here?
it's useful in some cases, but mostly it hides errors that we don't want to hide

---

I agree, was thinking of that when writing it. kept in, but there's no need, it's removed now